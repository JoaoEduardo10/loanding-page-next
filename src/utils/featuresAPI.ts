import Icon from '../../public/images/icon.svg';
import { CardsFaturesProps } from '../components/CardsFatures';

export type mockFeturesProps = CardsFaturesProps

export const mockFeatures: mockFeturesProps[] = [
	{
		icon: Icon,
		heading: 'Digital Strategy',
		text: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna.'
	},
	{
		icon: Icon,
		heading: 'Estrat√©gia digital',
		text: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.'
	},
	{
		icon: Icon,
		heading: 'Social Midia',
		text: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.'
	},
	{
		icon: Icon,
		heading: 'Social Midia',
		text: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.'
	},
];