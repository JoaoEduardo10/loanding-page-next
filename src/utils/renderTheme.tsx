import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { GlobalStyled } from '../styles/globalStyles';
import { theme } from '../styles/theme';

export const renderTheme = (children: React.ReactNode) => {
	return render(
		<ThemeProvider theme={theme}>
			{children}
			<GlobalStyled />
		</ThemeProvider>
	);
};