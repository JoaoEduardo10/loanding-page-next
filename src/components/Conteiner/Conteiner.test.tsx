import { screen } from '@testing-library/react';
import { Conteiner } from '.';
import { renderTheme } from '../../utils/renderTheme';

describe('<Conteiner />', () => {
	it('should render a Components Conteiner with  height 100%', () => {
		const { container } = renderTheme(<Conteiner>Children</Conteiner>);

		const conteiner = screen.getByLabelText('Conteiner');

		expect(conteiner).toHaveStyleRule('height', '100%');

		expect(container.firstChild).toMatchSnapshot();
	});

	it('should render a Components Conteiner with  height 100vh', () => {
		const { container } = renderTheme(<Conteiner height>Children</Conteiner>);

		const conteiner = screen.getByLabelText('Conteiner');

		expect(conteiner).toHaveStyleRule('height', '100vh');

		expect(container.firstChild).toMatchSnapshot();
	});
});