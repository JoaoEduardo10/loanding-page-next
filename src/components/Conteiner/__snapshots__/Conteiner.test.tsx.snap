// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Conteiner /> should render a Components Conteiner with  height 100% 1`] = `
.c0 {
  height: 100%;
  width: 100%;
  border-bottom: 1rem solid #222222;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<section
  aria-label="Conteiner"
  class="c0"
>
  Children
</section>
`;

exports[`<Conteiner /> should render a Components Conteiner with  height 100vh 1`] = `
.c0 {
  height: 100vh;
  width: 100%;
  border-bottom: 1rem solid #222222;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<section
  aria-label="Conteiner"
  class="c0"
>
  Children
</section>
`;
